src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java

```diff
@@ -706,6 +708,80 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    private void invokeAdd(Method m, Object[] args) throws ExecutionException {
+        if (params.isImmutable()) {
+            throw new IllegalStateException("Could not call set on immutable object");
+        }
+        String attrName = Character.toLowerCase(m.getName().charAt(5)) + m.getName().substring(6);
```

Comment:
The caller does not check that the given attribute is collection or not. Have to check, becuse not all attribute type can support `addTo` and `removeTo` method. Maybe this check can be placed when MapProxy is created and attributes are processed, but it's I think harder, because have to iterate all addTo removeFrom method an dget name and check after. So the check nicer in MapProxy createiom but this check is acceptable on call also.

Reply:
[b7d7975](https://github.com/BlackBeltTechnology/structured-map-proxy/pull/82/commits/b7d79756f1b886512710a06d2298613ff0d17eae)